geode-core/src/integrationTest/java/org/apache/geode/TxWithCacheLoderIntegrationTest.java

```diff
@@ -0,0 +1,140 @@
+import org.apache.geode.cache.LoaderHelper;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.RegionFactory;
+import org.apache.geode.cache.RegionShortcut;
+import org.apache.geode.cache.TransactionId;
+
+public class TxWithCacheLoderIntegrationTest {
+
+  private static final String REGION_NAME =
+      TxWithCacheLoderIntegrationTest.class.getSimpleName() + "_Region";
+  private static final String VALUE2 = "value2";
+  private final String KEY1 = "key1";
+  private final String KEY2 = "key2";
+  private static final String VALUE1 = "value1";
+
+  private Cache cache;
+  private CacheTransactionManager transactionManager;
+
+  private Region<String, String> region;
+
+  @Before
+  public void setUp() throws Exception {
+    cache = new CacheFactory().create();
+    transactionManager = cache.getCacheTransactionManager();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    if (cache != null) {
+      cache.close();
+    }
+    cache = null;
+    region = null;
+  }
+
+  protected Properties getConfig() {
+    Properties config = new Properties();
+    config.setProperty(LOCATORS, "");
+    config.setProperty(MCAST_PORT, "0");
+    return config;
+  }
+
+
+  @Test
+  public void txWithCacheLoaderThrowsIfThereIsConflict() {
+    region = createRegion(REGION_NAME);
+    // first operation
+    transactionManager.begin();
+    region.put(KEY1, VALUE1);
+    assertThat(region.get(KEY1)).isEqualTo(VALUE1);
+    TransactionId txId = transactionManager.suspend();
+
+    // second operation with cacheLoader
+    transactionManager.begin();
+    region.getAttributesMutator().setCacheLoader(new MyLoader());
+    assertThat(region.get(KEY1)).isEqualTo(KEY1);
+
+    transactionManager.commit();
+
+    // resume first transaction
+    transactionManager.resume(txId);
+    Throwable thrown = catchThrowable(transactionManager::commit);
+    assertThat(thrown).isInstanceOf(CommitConflictException.class);
+    assertThat(region.get(KEY1)).isEqualTo(KEY1);
+
+
+  }
+
+  @Test
+  public void txWithCacheLoaderSucceedsIfThereIsNoConflict() {
+    Region<String, String> region = createRegion(REGION_NAME);
+    // first operation
+    transactionManager.begin();
+    region.put(KEY2, VALUE2);
+    assertThat(region.get(KEY2)).isEqualTo(VALUE2);
+    TransactionId txId = transactionManager.suspend();
+
+    // second operation with cacheLoader
+    transactionManager.begin();
+    region.getAttributesMutator().setCacheLoader(new MyLoader());
+    assertThat(region.get(KEY1)).isEqualTo(KEY1);
+
+    transactionManager.commit();
+
+    // resume first transaction
+    transactionManager.resume(txId);
+    transactionManager.commit();
+    assertThat(region.get(KEY2)).isEqualTo(VALUE2);
+
+  }
+
+  private static class MyLoader implements CacheLoader {
+
+    @Override
+    public void close() {}
+
+    @Override
+    public Object load(LoaderHelper helper) throws CacheLoaderException {
```

Comment:
raw use of parameterized class can be solved as followings.
    @Override
    public Object load(LoaderHelper<Object, Object>  helper) throws CacheLoaderException {
      return helper.getKey();
    }